dataset/groupby.cpp

```diff
@@ -343,10 +358,18 @@ GroupBy<DataArray> groupby(const DataArrayConstView &array, const Dim dim) {
 GroupBy<DataArray> groupby(const DataArrayConstView &array, const Dim dim,
                            const VariableConstView &bins) {
   const auto &key = array.coords()[dim];
-  return {
-      array,
-      core::CallDType<double, float, int64_t, int32_t>::apply<MakeBinGroups>(
-          key.dtype(), key, bins)};
+  return callDType<DataArray, DataArrayConstView>(array, key, bins);
```

Comment:
Call the newly added overload for `groupby` here. But calling the templated `callDType` you added we suffer from (slightly) increased compilation times and binary sizes.